src/Paket.Core/Packaging/NupkgWriter.fs

```diff
@@ -0,0 +1,383 @@
+        let root = XElement(ns + "Types")
+
+        let defaultNode extension contentType =
+            let def = XElement(ns + "Default")
+            def.SetAttributeValue(XName.Get "Extension", extension)
+            def.SetAttributeValue(XName.Get "ContentType", contentType)
+            def
+
+        let knownExtensions =
+            Map.ofList [ "rels", "application/vnd.openxmlformats-package.relationships+xml"
+                         "psmdcp", "application/vnd.openxmlformats-package.core-properties+xml" ]
+
+        let ext path = Path.GetExtension(path).TrimStart([| '.' |]).ToLowerInvariant()
+
+        let fType ext =
+            knownExtensions
+            |> Map.tryFind ext
+            |> function
+            | Some ft -> ft
+            | None -> "application/octet"
+
+        let contentTypes =
+            fileList
+            |> Seq.choose (fun f ->
+                   let e = ext f
+                   if String.IsNullOrWhiteSpace e then
+                     None
+                   else Some(e, fType e))
+            |> Seq.distinct
+            |> Seq.iter (fun (ex, ct) -> defaultNode ex ct |> root.Add)
+
+        XDocument(declaration, box root)
+
+    let nuspecDoc (info:CompleteInfo) =
+        let core,optional = info
+        let declaration = XDeclaration("1.0", "UTF-8", "yes")
+        let ns = XNamespace.Get "http://schemas.microsoft.com/packaging/2011/10/nuspec.xsd"
+        let root = XElement(ns + "package")
+
+        let addChildNode (parent : XElement) name value =
+            let node = XElement(ns + name)
+            node.SetValue value
+            parent.Add node
+
+        let metadataNode = XElement(ns + "metadata")
+        root.Add metadataNode
+        let (!!) = addChildNode metadataNode
+
+        let (!!?) nodeName strOpt =
+            match strOpt with
+            | Some s -> addChildNode metadataNode nodeName s
+            | None -> ()
+
+        let buildFrameworkReferencesNode libName =
+            let element = XElement(ns + "frameworkAssembly")
+            if String.IsNullOrEmpty libName then () else
+                element.SetAttributeValue(XName.Get "assemblyName", libName)
+            element
+
+        let buildFrameworkReferencesNode frameworkAssembliesList =
+            if List.isEmpty frameworkAssembliesList then () else
+            let d = XElement(ns + "frameworkAssemblies")
+            frameworkAssembliesList |> List.iter (buildFrameworkReferencesNode >> d.Add)
+            metadataNode.Add d
+
+        let buildDependencyNode (Id, requirement:VersionRequirement) =
+            let dep = XElement(ns + "dependency")
+            dep.SetAttributeValue(XName.Get "id", Id)
+            let version = requirement.FormatInNuGetSyntax()
+            if String.IsNullOrEmpty version then
+                dep.SetAttributeValue(XName.Get "version", "0.0")
+            else
+                dep.SetAttributeValue(XName.Get "version", version)
+            dep
```

Comment:
I hand-crafted @forki's fix in here because git wouldn't recognize this file anymore :)